Don't catch exception when resolving a class

passioneight opened the pull request:
For example, when using a `LinkGenerator`, the error message gets logged and that's about it.

This change will let the application throw an exception, so that any invalid reference used displays an error message right away.

passioneight committed:
Don't catch exception when resolving a class

For example, when using a `LinkGenerator`, the error message gets logged and that's about it.

This change will let the application throw an exception, so that any invalid reference used displays an error message right away.

dvesh3 commented:
BC Break? As existing implementations with missing classes will start throwing exception.

passioneight commented:
Fair point, though, I'm not sure if there is any use case where the developer just left it as is when `null` was returned. But it could be. May be better off in v11?

dpfaffenbauer committed:
Stop further processing of the video thumbnail if ffmpeg fails to generate it

fashxp committed:
fixed #12229 - [Task] Add @param and @return php doc types to /lib/Model

blankse committed:
Objectbrick: Allow localized block fields

dvesh3 committed:
Apply php-cs-fixer changes

aashan10 committed:
Added @param and @returns to docblocks for models/Asset (#12306)

kingjia90 committed:
Apply php-cs-fixer changes

kingjia90 committed:
[Task][Permissions] IsAllowed considering system permissions and adding related tests (#12316)

* added general permission unchecked tests

* fix wrong exception

* ultimately the kernel, protected checkPermission lead to this isAllowed

* new testhelper to call private and protected method

* args is array

* isAllowed checking master permission

* re-introduces isAllowed test for user5,6

* fixes some null approach leftover

* fixing user5 islallowed listing permission

* use null to assert master permission errors

* move check to abstractelement, change to assertException

* change to null for master permission assertion

* extend changes to areAllowed, added upgrade notes

* same permissions as isAllowed when $user is null

aryaantony92 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Fix Phpstan errors (#12383)

kingjia90 committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

kingjia90 committed:
locationawareconfigrepo using $_SERVER
$_ENV fallback in bootstrap when dotEnv is not set

dpfaffenbauer committed:
Static Class Builder should not call "classSaved" (#12376)

* Class Builder should not call "classSaved"

* Class Builder should not call "classSaved"

* Class Builder should not call "classSaved"

brusch committed:
Apply php-cs-fixer changes

aryaantony92 committed:
Apply php-cs-fixer changes

kingjia90 committed:
[Symfony] Fix some deprecations from version 5 (#12304)

* removing securityfactorInterface

https://github.com/symfony/symfony/blob/5.4/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SecurityFactoryInterface.php#L22

* Intlformatter using polyfill, added composer requirement

* UsernameNotFoundException replaced by UserNotFoundException in 5.3

https://github.com/symfony/security-core/blob/5.3/Exception/UsernameNotFoundException.php

* Anonymous is represented by absence of token

https://github.com/symfony/security-core/blob/4540ecb8ae82cc46d9580672888597f481ff0440/Authentication/Token/AnonymousToken.php#L21

* replacing with nulltoken (?)

* bumped security bundle to 5.4 to use addAuthenticatorFactory

* added upgrade notes for `UsernameNotFoundException`

dvesh3 committed:
[Elements] Fix getUserPermissions - follow up to #12316 (#12394)

kingjia90 committed:
Apply php-cs-fixer changes

kingjia90 committed:
prevent fatal error when getCurrentUser is null in getUserPermissions (#12395)

mcop1 committed:
[Task] Add @param and @return php doc types to /models/Element (#12298)

kingjia90 committed:
Apply php-cs-fixer changes

GALCF committed:
CMS Feature: Brick Preview Tooltip (#12343)

* Move brick icon resource-resolve to more generalized method

* Markdown table formatting

* [Feature] Add optional brick preview image tooltip

* Simplify brick preview logic, use plain html option

* Revert icon autoresolve refactor

kingjia90 committed:
[Improvement]: extend PIMCORE_CLASS_DEFINITION_WRITABLE to be globally read-only (#12392)

* wip writable env

* class/custom layout add button disabled based on ENV

* fixes a typo in docs

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* ENV var description in a bullet list
removing TODO comment

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

dvesh3 committed:
Apply php-cs-fixer changes

blankse committed:
Fix: Bug isAllowed() imploding parentIds with unsaved objects (id:null) (#12397)

dvesh3 committed:
Apply php-cs-fixer changes

blankse committed:
Fix: "Warning: Undefined array key "key"" (#12400)

* Fix: "Warning: Undefined array key "key""
* Check the key first, then the permission
* Permissions should be integers
* Optimize phpdoc

aashan10 committed:
Deprecated `RuntimeCacheTrait` and replaced its usage (#12385)

* Added deprecation notice on RuntimeCacheTrait

* Replaced the usage of RuntimeCacheTrait as it brought more problems than it solved

* Added  flag on the places that were using RuntimeCacheTrait

* Added params in docblock for newly added  flag

* Added deprecation warnings for RuntimeCacheTrait

* Update lib/Cache/RuntimeCacheTrait.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Removed deprecation replacement usage message

* Added deprecation notice to the release notes

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

robertSt7 committed:
disable "Visible in Grid View" and "Visible in Search Result" for CS in Class conf

blankse committed:
Fix: Error thrown while handling message Pimcore\Messenger\AssetUpdateTasksMessage (#12390)

* Fix: Error thrown while handling message Pimcore\Messenger\AssetUpdateTasksMessage

* Update AssetUpdateTasksHandler.php

Commit:
[Grid exports] Initially enable inheritance (#12325)

kingjia90 committed:
Apply php-cs-fixer changes

robertSt7 committed:
add @param and @return (#12380)

fashxp committed:
Apply php-cs-fixer changes

aryaantony92 committed:
Apply php-cs-fixer changes

michaelhader committed:
Fix updating application_logs table entries in LogMailMaintenanceTask by also checking if 'maintenanceChecked' column is NULL (#12389)

robertSt7 committed:
[Bug]: Cleanup FC and Bricks Stores on lower-case DB Servers (#12391)

* Ignore case sensitivity in id

* Update tests/Model/DataObject/DefinitionModifierTest.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update tests/Model/DataObject/DefinitionModifierTest.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

aashan10 committed:
Added @params  and @returns to docblocks to lib/Targeting (#12319)

* Added @params  and @returns to docblocks

* Fixed phpStan issue

* Added missing @param and @returns docblocks

robertSt7 committed:
Apply php-cs-fixer changes

jdreesen committed:
Fix project root determination case in `pimcore-install`

aryaantony92 committed:
[Task] Add @param and @return php doc types to /lib/Workflow (#12401)

* [Task] Add @param and @return php doc types to /lib/Workflow

* Type casted return value to boolean

robertSt7 committed:
Apply php-cs-fixer changes

blankse committed:
ClassDefinition: maxItems property should be int or null (#12354)

* ClassDefinition: maxItems property should be int or null

* Fix doc types with getAsIntegerCast/getAsFloatCast usage

dvesh3 committed:
[Task] Add @param and @return php doc types to /lib/<REST (#12382)

kingjia90 committed:
Apply php-cs-fixer changes

jdreesen committed:
Fix some typos in the PHPDoc description of events (#12425)

timwiechers committed:
Add JavaScript events for "preDelete" and "postDelete" for Assets, Documents and Data objects. (#11866)

* Add JavaScript events for "preDelete" and "postDelete" for Assets, Documents and Data objects.

* Updated docs for JS events for "preDelete" and "postDelete" for Assets, Documents and Data objects.

* Update bundles/AdminBundle/Resources/public/js/pimcore/elementservice.js

Co-authored-by: Dennis Pfahlbusch <22637136+dpfahlbusch@users.noreply.github.com>

* fixing pre-delete

Co-authored-by: Dennis Pfahlbusch <22637136+dpfahlbusch@users.noreply.github.com>
Co-authored-by: Christian Fasching <fashxp@users.noreply.github.com>

brusch committed:
Cookies should be auto-secure (#12405)

* Cookies should be auto-secure - resolves #12377

* clearCookie() needs $secure = null (default is different from Cookie::__construct())

kingjia90 committed:
[Task] Add @param and @return php doc types to /lib/Twig (#12396)

* fix @param and @return php doc types to /lib/Twig

* added missing $placement type

mcop1 committed:
Apply php-cs-fixer changes

blankse committed:
Fix: Error with handling AssetPreviewImageMessage

brusch committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

Johanna-Maria committed:
[Ecommerce] Add Event to GetFields-Method of IndexController (#12312)

* Add Event for getFilters (IndexController)

Sometimes, it would be useful to exclude/change some of the fields returned by the getFieldsAction method of the IndexController. I propose dispatching an event (like it already happens within the getValuesForFilterFieldAction method).

* add constant to AdminEvents

* change constant name

* Apply suggestions from code review

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

brusch committed:
XLIFF: fixing import when working with html entities - see #11859

mcop1 committed:
[Security] SQL Injection in Data Hub GraphQL (#12444)

* [Security] SQL Injection in Data Hub GraphQL (AbstractListing)

* Update lib/Model/Listing/AbstractListing.php

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Update lib/Model/Listing/AbstractListing.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>
Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

brusch committed:
Apply php-cs-fixer changes

aashan10 committed:
Fixed classification group remove triggering sql error bug

mcop1 committed:
[Bug][Backend]: Wrong Date and DateTime value - adding timezone configuration to the docs

ultramegatom committed:
Children grid doesn't open when there are no select options

blankse committed:
Don't cache thumbnail urls with error image (#12434)

* Don't cache thumbnail urls with error image

* Fix php doc

brusch committed:
Apply php-cs-fixer changes

robertSt7 committed:
[Bug]: Fix batch append/remove doesn't work on CS attributes (#12418)

* fix Batch append for CS-multiselect

* Update bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dpfaffenbauer committed:
Fix FFMPEG with durations like 00:00:11.04 (milliseconds) (#12449)

* Fix FFMPEG with durations like 00:00:11.04 (milliseconds)

* getDuration() can return null

* Update lib/Video/Adapter/Ffmpeg.php

Co-authored-by: Sebastian Blank <sebastian.bl@gmx.de>

* Update Adapter.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>
Co-authored-by: Sebastian Blank <sebastian.bl@gmx.de>

kingjia90 committed:
moved block as only system field could be columns (#12447)

that may need the quoteIndentifier

blankse committed:
Fix: Error in GeneratePagePreviewHandler (#12439)

* Fix: Error in GeneratePagePreviewHandler

* Chromium::convert() should also return false if outputFile couldn't save

* Add debug log for convert failure

brusch committed:
Use Symfony Config to init Doctrine connection

dpfaffenbauer committed:
[PHP 8.1] fix Passing null to parameter #1 ($string) of type string is deprecate (#12458)

* [PHP 8.1] fix Passing null to parameter #1 ($string) of type string is deprecate

https://github.com/coreshop/CoreShop/runs/6981738951?check_suite_focus=true#step:17:93

* Update Staticroute.php

* Update Staticroute.php

* Update Staticroute.php

* Update models/Staticroute.php

Co-authored-by: Sebastian Blank <sebastian.bl@gmx.de>

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>
Co-authored-by: Sebastian Blank <sebastian.bl@gmx.de>

dvesh3 committed:
Apply php-cs-fixer changes

brusch committed:
[Thumbnails] Add width, height and filesize to the cache (#12461)

* perf: add file size, width and height to thumbnail cache

* simplifications and added cache for getFilesize()

Co-authored-by: Dalibor Karlović <dalibor.karlovic@sigwin.hr>

brusch committed:
Apply php-cs-fixer changes

dvesh3 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Merge branch '10.4' into 10.x

dvesh3 committed:
Merge remote-tracking branch 'origin/10.x' into 10.x

dvesh3 committed:
[Core] Migrate Custom Layout definition files to use LocationAwareConfigRepository  (#12428)

kingjia90 committed:
Apply php-cs-fixer changes

dpfahlbusch committed:
[IMPROVEMENT] add min, max, decimalsize and -precision to quantityValue (#12368)

dvesh3 committed:
Fix Error when removing classification group which has key relationships - resolves #11284

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

dvesh3 committed:
[Admin] Fix UI does not recover from maintenance mode - resolves 12372 (#12426)

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

jdreesen committed:
Don't read package info from composer.lock

kingjia90 committed:
[Improvement]: Remove/deprecate some folder creation by MkdirCacheWarmer (#12443)

* removing PIMCORE_LOG_MAIL_PERMANENT (follow-up #11366)

* removing PIMCORE_LOG_FILEOBJECT_DIRECTORY
since it can be created in the FileObject constructor
https://github.com/pimcore/pimcore/blob/cae32ab5dea2e5237a798cdfb6b8f18266d9a72b/lib/Log/FileObject.php#L39

* mkdir PIMCORE_CLASS_DEFINITION_DIRECTORY based on PIMCORE_CLASS_DEFINITION_WRITABLE

* PIMCORE_CLASS_DIRECTORY also related to PIMCORE_CLASS_DEFINITION_WRITABLE ?

* looping only unique folders

* Update lib/HttpKernel/CacheWarmer/MkdirCacheWarmer.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

MarcW09 committed:
[Improvement] add config data for area and image editable

dpfaffenbauer committed:
Glossary: allow html entities like &nbsp; or &copy; (#12483)

* Glossary: allow html entities like &nbsp; or &copy;

* Update Processor.php

* Glossary: allow html entities like &nbsp; or &copy;

* Glossary: allow html entities like &nbsp; or &copy;

dvesh3 committed:
Apply php-cs-fixer changes

jdreesen committed:
Add DirectoryAwareVersionComparator

jdreesen committed:
Fix typo in the docs

dpfahlbusch committed:
[Improvement] add name to export / set name of custom layout on import if na… (#12420)

* [TASK] add name to export / set name of custom layout on import if name is available in custom layout json file

Resolves: #5023

* [TASK] make error message more speakable

Relates #5023

* [TASK] fix merge conflicts

Relates #5023

* [TASK] fix merge conflicts

Relates #5023

* [TASK] fix tests

Relates #5023

* Update bundles/AdminBundle/Resources/public/js/pimcore/object/helpers/customLayoutEditor.js

Co-authored-by: aryaantony92 <97134765+aryaantony92@users.noreply.github.com>

* Update bundles/AdminBundle/Controller/Admin/DataObject/ClassController.php

Co-authored-by: aryaantony92 <97134765+aryaantony92@users.noreply.github.com>

Co-authored-by: aryaantony92 <97134765+aryaantony92@users.noreply.github.com>

dvelguru committed:
Fix: toggle tag highlighting check if element is not null

dpfaffenbauer committed:
[Db] remove all usages of Pimcore\Db\ConnectionInterface and Pimcore\Db\PimcoreExtensionstrait (#12123)

* [Db] deprecate Pimcore\Db/Connection and change return types in Pimcore\Connection\Db

* [Db] remove usages of escapeLike

* [Db] remove usages of fetchCol

* [Db] remove usages of fetchRow

* [Db] remove usages of fetchRow and fetchCol

* [Db] remove usages of updateWhere

* [Db] remove usages of deleteWhere

* [Db] remove usages of insertOrUpdate

* [Db] remove usages of insertOrUpdate

* [Db] remove usages of escapeLike

* [Db] remove usages of quoteInto

* [Db] remove usages of quoteInto

* [Db] remove usages of quoteInto

* [Db] remove usages of queryIgnoreError

* [Db] remove usages of fetchOne

* [Db] remove usages of fetchOne

* [Db] remove usages of selectAndDeleteWhere

* [Db] fix remaining stan issues

* [Db] remove all internal usages of ConnectionInterface

* [Db] remove all internal usages of ConnectionInterface

* [Db] fix remaining stan issues

* [Db] revert DB changes for BC

* [Db] revert DB changes and make BC

* Update ConnectionInterface.php

* Update PimcoreExtensionsTrait.php

* commit suggestions

* Update bundles/AdminBundle/EventListener/ImportConfigListener.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

* commit suggestions

* commit suggestions

* Update lib/Model/Listing/AbstractListing.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

brusch committed:
Apply php-cs-fixer changes

brusch committed:
Fixed installer (#12545)

* Fixed installer - regression of #12123

* make PhpStan happy again 💘

brusch committed:
Fix installer

brusch committed:
Activate Stale action

dpfaffenbauer committed:
Glossary fix for &lt; &gt; and &amp; (#12541)

* [Glossary] fix for &amp; and &nbsp;

* [Glossary] another test for &lt; and &gt;

mattamon committed:
fix: Ecommerce broken Index Field Selection (#12488)

* fix: Ecommerce broken Index Field Selection

 Resolves #12487
Check if row is false so nullable values do not brake the loop

* fix: php stand static code analysis 

fetch row explicitly and check if row !== false

dvesh3 committed:
Merge branch '10.4' into 10.x

dvesh3 committed:
Fix merge conflict - related to https://github.com/pimcore/pimcore/pull/12502s

dvesh3 committed:
Deprecate Wkhtmltopdf - resolves #12431

questionmark78 committed:
Fix: Exception when using Required By Operator in Grid Options (#12493)

robertSt7 committed:
[Bug]: Cannot filter on multiple CS keys on object grid  (#12499)

* fix filter on multiple cs keys in grid

* fix empty filtering

* Update bundles/AdminBundle/Helper/GridHelperService.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

aryaantony92 committed:
[Improvement]: Refactor navigation builder to use param style arguments (#12452)

* Refactor navigation builder to use param style arguments

* php stan fix

* Array indexing Fix

* Refactor navigation builder to use param style arguments - #12310

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* Add upgrade Note

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Update doc

* Update doc and allowed string type

Co-authored-by: Divesh <divesh.pahuja@pimcore.com>
Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

aryaantony92 committed:
Apply php-cs-fixer changes

mcop1 committed:
Error message while selecting logs in LogController  (#12564)

* Changed fetchAssociative to fetchAllAssociative to get all logs again

* Revert "Changed fetchAssociative to fetchAllAssociative to get all logs again"

This reverts commit fa7ea681a6ee970f5dc352043a9d2729dd45d56a.

* Changed fetchAssociative to fetchAllAssociative to get all logs again

timwiechers committed:
Add an option "allowedTypes" to the video editable. (#12445)

* Add an option "allowedTypes" to the video editable.

* Prevent method getVideoType from returning null. Code cleanup.

* Prevent not allowed types from being used.

dvesh3 committed:
Apply php-cs-fixer changes

kingjia90 committed:
[Improvement] Application Logger File Objects should be stored via flysystem storage (#12460)

* wip moving application logger file obj to flysystem

* changed to local like others

* removing PIMCORE_LOG_FILEOBJECT_DIRECTORY

* further fixes and improved error

* delete file list from archived rows
this choice is to avoid expensive recursive listContents, inability for last modification date check for cloud storages, downside is that no more limited to be ran between midnight and 4a.m.
also added a folder deleter for the motnhly archivation

* remove obsolete lock and moved folder archivation

* adding some changelog notes

* Resolved merge conflicts

Co-authored-by: mcop1 <marco.perberschlager@pimcore.com>

jdreesen committed:
Support modern bundle structure (#12356)

* WIP

* Take modern bundle structure into account

* Take modern bundle structure into account

* Take modern bundle structure into account

* Update comment

* Take modern bundle structure into account

dvesh3 committed:
Apply php-cs-fixer changes

Fahl-Design committed:
Fix: #12497 - cast date string to timestamp to fix date compare (#12498)

Signed-off-by: Benjamin Fahl <git@fahl-design.de>

MarkTro committed:
Change the function "getClosestParentOfClass" to from "private" to "public" (#12546)

I found this method always very useful if you need a specific parent type. Therefore I would like to have it "public" again. Otherwise I need to implement it myself, which would be mostly just copy and paste.

aashan10 committed:
Deprecated MissingDependencyException (#12538)

JesusValeraDev committed:
Fix: Solve typo when outputting a DataObject class (#12506)

dvelguru committed:
Refactor: folder properties declared twice (#12573)

naveensrinivasan committed:
[Actions] Added PR Checklist action

naveensrinivasan committed:
chore: Set permissions for GitHub actions (#12451)

* chore: Set permissions for GitHub actions

 Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

* Update .github/workflows/stale.yml

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

vicefi committed:
Update 42_Scheduled_Block.md

mcop1 committed:
Fix: Merge dashboards from local config and perspectives (#12578)

Added disabledPortlets to configuration

robertSt7 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Fixed variant filter - resolves #12378

aryaantony92 committed:
Disabled the option to drag and drop objects if the target is a paginated tree (#12555)

* Disabled drag and drop objects if the target is a paginated tree

* Added message for advanced option

* Update bundles/CoreBundle/Resources/translations/en.json

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
[Translations] Filter by languages (#12539)

* [Translations] Filter by languages - resolves #12490

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Translations] Filter by languages - add test resolves #12490

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* Update tests/Unit/Translation/TranslatorTest.php

Co-authored-by: robertSt7 <104770750+robertSt7@users.noreply.github.com>

* [Translations] Filter by languages - review changes #12490

Co-authored-by: robertSt7 <104770750+robertSt7@users.noreply.github.com>

dturopoli committed:
[FIX]: Filter for object brick attributes does not work (#12416)

* Object brick grid filter/sort fix

* Fix localized field d&d in grid

robertSt7 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Translations] Fix memory issue while importing large data (#12547)

* [Translations] Fix timeout when importing large data - resolves #11082

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Translations] Fix timeout when importing large data - resolves #11082

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* Update models/Translation.php

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

MercuryKojo committed:
getHttpData() throws error without query params

an isset validation is needed before merging the get params of the url to not throw an exception in php8.1

brusch committed:
Apply php-cs-fixer changes

robertSt7 committed:
Change JS core events for bundles to native JS events (#12566)

* add native events for broker

* replace event-dispatcher.js with native events

* add new line at end of file

MercuryKojo committed:
addCondition default $value value should be array (#12562)

* addCondition default $value value should be array

if you don't give a value for addCondition the function calls setParameters with [null] which gives an "SQLSTATE[HY093]: Invalid parameter number: number of bound variables does not match number of tokens" Error
So the default value should be [], since setParameters requires array anyway

* Update bundles/EcommerceFrameworkBundle/OrderManager/Order/Listing.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

aryaantony92 committed:
Apply php-cs-fixer changes

Stefan-JustMe69-SQ committed:
12569 - Fix. Ignore classes === null. (#12587)

* #12569 - Fix. Ignore classes === null.

* #12569 - Fix. Ignore classes === null. Change param definition and return type.

* #12569 - Fix. Ignore classes === null. Change param definition and return type.

* #12569 - Fix. Ignore classes === null. Change param definition and return type.

fashxp committed:
Fixed object reference setting for objectbricks and fieldcollections … (#12588)

* Fixed object reference setting for objectbricks and fieldcollections when loading data from cache

* fixed phpstan

* Update models/DataObject/Concrete.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

brusch committed:
Apply php-cs-fixer changes

aryaantony92 committed:
Disable drag and drop for objects with manual indexes (#12602)

* Disable drag and drop for objects with manual indexes

* Update bundles/CoreBundle/Resources/translations/en.json

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Update bundles/AdminBundle/Resources/public/js/pimcore/object/tree.js

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* update Translations

* Update bundles/CoreBundle/Resources/translations/en.json

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Deprecate legacy security authentication classes - resolves #12553

dpfaffenbauer committed:
Allow Glossary Processor BlockedTags to be configurable (#12512)

* Allow Glossary Processor BlockedTags to be configurable

* Update bundles/CoreBundle/Resources/config/pimcore/default.yaml

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

mcop1 committed:
Merge branch '10.4' into 10.x

Brainshaker95 committed:
Add getObjects to annotated class methods inside of generated listing classes (#12489)

dvesh3 committed:
[Session] Do not use deprecated session service injections (#12241)

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

* [Session] Do not use deprecated session service injections - #11042

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* fixed SessionEnvironment

* [Session] Do not use deprecated session service injections - #11042

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Do not use deprecated session service injections -review changes #11042

* [Session] Do not use deprecated session service injections - #11042

Co-authored-by: Christian Fasching <christian.fasching@pimcore.com>

fashxp committed:
Apply php-cs-fixer changes

jdreesen committed:
Improve navigation builder a bit (followup of #12452) (#12565)

* Improve navigation builder a bit

* Improve unpacking options in navigation builder

brusch committed:
Added PR checklist action

brusch committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

# Conflicts:
#	.github/workflows/pull-request-checklist.yml

aashan10 committed:
[Improvement]: Rename Pimcore\Cache\Runtime to Pimcore\Cache\RuntimeCache (#12548)

* Added RuntimeCache with BC layer for Cache\Runtime

* Replaced deprecated \Pimcore\Cache\Runtime with \Pimcore\Cache\RuntimeCache

* Do not use deprecated Runtime class - #12474

* Do not use deprecated Runtime class - #12474

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

dvesh3 committed:
Apply php-cs-fixer changes

timwiechers committed:
Fix translation for reordering block items. (#12611)

dvesh3 committed:
[Session] Deprecate session configurators in favor of event listeners (#12294)

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Do not use deprecated session service injections - #11042

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* [Session] Deprecate session configurators in favor of event listeners - review changes - related to #12241

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

* [Session] Deprecate session configurators in favor of event listeners - related to #12241

fashxp committed:
Apply php-cs-fixer changes

fashxp committed:
fix possible key not existing error (#12612)

thijs-windels committed:
Fix document relations editable when adding multiple objects

When selecting multiple objects in document the relations editable,
only the first one was added. Fixed this to be able to add multiple.

dvesh3 committed:
Revert "[Core] Migrate Custom Layout definition files to use LocationAwareConfigRepository(#12428) (#12543)

* Revert "[Core] Migrate Custom Layout definition files to use LocationAwareConfigRepository  (#12428)"

This reverts commit 62a46881

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

* Revert "[Core] Migrate Custom Layout definition files to use LocationAwareConfigRepository  (#12428)"

This reverts commit 62a46881

Signed-off-by: Divesh <divesh.pahuja@pimcore.com>

dvesh3 committed:
[Task] Add @param and @return php doc types to /models/<REST> (#12339)

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* Update models/WebsiteSetting/Listing.php

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* Changed Recipient, title and message fields to not null since it cannot be empty

* Changed type hint of getRecipient() to not null

* Removed ternary check to avoid php stan error

* php stan fix

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* [Task] Add @param and @return php doc types to /models/<REST> - resolves #12215

* Fix Typo

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>
Co-authored-by: aryaantony92 <arya.antony@pimcore.com>

aryaantony92 committed:
Apply php-cs-fixer changes

brusch committed:
PR checklist action fix (#12620)

* PR checklist action fix

* PR checklist action fix

* PR checklist action fix

brusch committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

rberneder committed:
fix: video file extension check should not be case-sensitive

jdreesen committed:
Fix root cause of #12569 (followup of #12587) (#12608)

* Remove non-existing classes in return value of AbstractObject\Dao::getClasses()

* Revert changes of #12587

robertSt7 committed:
[Data Objects]: Deprecate "generate type declarations" in class definitions (#12567)

* mark generate type declaration as deprecated

* Docs: Add upgrade notes for deprecate generate type declarations

* Update doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

brusch committed:
PR checklist rename

brusch committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

dvesh3 committed:
[Docs] Content Security Policy - Add WYSIWYG warning - related to #12594

dvesh3 committed:
Follow up to #12550

dvesh3 committed:
[BackOffice] Fix page count - resolves #12605 (#12626)

kingjia90 committed:
Fix: escapes the background url that may contain % symbols that can get confused as parameters by symfony

brusch committed:
[Elements] change $force parameter to a generic $params array (#12515)

* [Elements] change $force parameter to a generic $params array

* [Elements] change $force parameter to a generic $params array

* Update models/Element/AbstractElement.php

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Update doc/Development_Documentation/26_Best_Practice/95_Working_With_Runtime_Cache.md

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Fixed tests

* Fixed tests

* Fixed tests & deprecations

* Fixed tests & deprecations

* only issue deprecation if $force param was set

* fixed default value

* fixed default value

* fixed deprecation messages from getByPath()

* fixed deprecation messages

* fixed method call

* Fixed Test

* Add Upgrade Notes

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>
Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>
Co-authored-by: aryaantony92 <arya.antony@pimcore.com>

aryaantony92 committed:
Apply php-cs-fixer changes

brusch committed:
[Cache] DataObject classes warmer more independent, not requiring database (#12577)

kingjia90 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Docs] Apache Configuration - add htaccess rewrite rules in docs (#12619)

* [Docs] Apache Configuration - add htaccess rewrite rules in docs

* [Docs] Apache Configuration - add htaccess rewrite rules in docs

dvesh3 committed:
[Docs] Apache Configuration - add htaccess rewrite rules in docs (#12619)

* [Docs] Apache Configuration - add htaccess rewrite rules in docs

* [Docs] Apache Configuration - add htaccess rewrite rules in docs

aashan10 committed:
Replaced deprecated `Doctrine\DBAL\Connection` methods with recommended methods (#12622)

* Replaced deprecated  method with

* Fixed PHPStann issues

* Replaced deprecated query method with executeQuery

* Removed non-reachable ignored phpstan error message

* Replaced deprecated method with its recommended counterpart

* Replaced deprecated methods with suggested methods

timwiechers committed:
Update docs to mention both translation lists for POEditor.

dvesh3 committed:
Update baseline - remove ignored error pattern (#12640)

* Update baseline - remove ignored error pattern

* Fix phpstan errors

dvesh3 committed:
Merge branch '10.4' into 10.x

# Conflicts:
#	bundles/EcommerceFrameworkBundle/OrderManager/AbstractOrderList.php
#	models/Document/Editable/Video.php

dvesh3 committed:
Improve upgrade notes

jdreesen committed:
Allow disabling of marking the active trail when building the navigation (#12572)

* Allow disabling the marking of the active trail

* Refactor Builder::markActiveTrail() a bit

jdreesen committed:
[Improvement]: dispatch an event after an element was loaded (#12511)

* Dispatch an event after an element was loaded

* Add $params of getById() as event argument

brusch committed:
Deprecate Pimcore\Cache\Runtime class as well - follow up to #12548

dvesh3 committed:
Apply php-cs-fixer changes

robertSt7 committed:
[Improvement]: Replace plugins with event listener (#12637)

* Improve: Replace plugins with event listener

* add new line at eof

* Improve: Plugin deprecated warning

* Revert "Improve: Plugin deprecated warning"

This reverts commit aa262652394858ef431b0ad025f95b99629d04fd.

* Improve: optimize variable declarations

dvesh3 committed:
Fix Symfony deprecations  (#12635)

Load Legacy authenticator services when only when authenticator.manager is disabled 
Do not pass deprecated $authenticationManager argument to AuthorizationChecker
Do not call deprecated method AbstractAuthenticator::createAuthenticatedToken 
UserProvider must implement loadUserByIdentifier
DoctrineDbalStore instead of PdoStore for Symfony Locks 
Twig through DI instead of accessing directly by container anymore

kingjia90 committed:
Apply php-cs-fixer changes

mcop1 committed:
[CustomLayouts] Add filter option on listing (#12641)

* Feature: Filter options for custom layout listing.
Moved filter from JsonListing to a new trait.

* Fixed filtering id/classid

* Added FilterListingInterface, OrderListingInterface and OrderListingTrait.php
Refactored existing classes to use interface + traits instead of JsonListing
Deprecated JsonListing class.

* Keep extending from JsonListing for bc reasons
Renamed interfaces

robertSt7 committed:
Apply php-cs-fixer changes

robertSt7 committed:
Add upgrade notes and docs for JS events (#12633)

* Docs: Add upgrade notes for event listener
 Add migration docs for plugin to event listener
 Update all docs from plugin to event listener

* Feat: Show console warning, when a plugin is used

* Docs: Replace plugin with event listener

* Improve: Plugin deprecated warning

* Update bundles/AdminBundle/Resources/public/js/pimcore/plugin/broker.js

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update doc/Development_Documentation/20_Extending_Pimcore/13_Bundle_Developers_Guide/06_Event_Listener_UI.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update doc/Development_Documentation/23_Installation_and_Upgrade/07_Updating_Pimcore/README.md

Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

* Docs: Update to preparing for V11

* Docs: move event descriptions to events.js
      optimize variable declarations
      add link to event listener in upgrade notes

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>
Co-authored-by: Jacob Dreesen <j.dreesen@neusta.de>

dvesh3 committed:
[Admin] Add delete confirm dialog boxes (#12643)

* [Admin] Add delete confirm dialog boxes - resolves #6259

* [Admin] Add delete confirm dialog boxes - resolves #6259

* [Admin] Add delete confirm dialog boxes - resolves #6259

* Translation entry for pricing rule

Co-authored-by: aryaantony92 <arya.antony@pimcore.com>

NiklasBr committed:
Clarified doc comment in Processor.php

kingjia90 committed:
[Bug]: Fix missing $force param that includes published in GetLatestVersion()  (#12662)

* Fix: reintroduced the former $force parameter to include the current version when using latestVersion

* Task: renamed parameter name to $includingPublished

* Fix: phpdoc param renamed

Co-authored-by: Niklas <niklas.brunberg@cag.se>

NiklasBr committed:
Only generate webv2print PDF file $path when required

NiklasBr committed:
Adds missing translations for web2print_headlesschrome_*

kingjia90 committed:
[Bug]: Fix document PHP API not saving versionNote (#12652)

* Fix: versionNote was not properly passed to the parent function

* Test: added test about saving version notes

* Test: what happens if i remove this line?

* Test: wrong line

* Test: Added same tests for assets and objects, some minor text/comment improvement

* Test: changed to use getLatestVersion (requires #12662 to pass)

* Test: removed obsoleted $versions

dvesh3 committed:
Apply php-cs-fixer changes

brusch committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

# Conflicts:
#	phpstan-baseline.neon

brusch committed:
Fixed PhpStan merge error

kingjia90 committed:
[Improvement]: Deprecate legacy \Pimcore\Config\Config (#12650)

* Task: removing outdated removed method from documentation, see #8859

* Task: added deprecated tag in phpdoc block

* Task: removed a commented line about Pimcore\Config\Config

* Docs: added line about deprecation

brusch committed:
Improved docs for thumbnail prefixes - fixes #11691

jdreesen committed:
Fix generated PhpDoc parameter type of many-to-one relation fields (#12646)

dvesh3 committed:
Follow up to 12641 (#12670)

* Follow up to 12641 - Fix CustomLayout\Listing & Property\Predefined\Listing

* Follow up to 12641 - Fix CustomLayout\Listing

* Follow up to 12641 - Fix CustomLayout\Listing

* Follow up to 12641 - Fix CustomLayout\Listing

* Follow up to 12641 - Fix CustomLayout\Listing

mcop1 committed:
Apply php-cs-fixer changes

kingjia90 committed:
[Bug]: Prevent Document Link targeting itself causing redirection loop (#12625)

* Fix: implemented some exception check when the link dropped/typed is the same as the link itself during the save

* Fix: disabling drag and drop of link itself and added error when selected through search

* Task: moved check to model, removing from controller

* Test: trying to set link target id as itself without crashing in inf. loop

* Test: added debug output line to hint in case of circular reference problem

* Fix: moved check before loading object

dvesh3 committed:
Apply php-cs-fixer changes

dpfaffenbauer committed:
Fix ObjectBrickContainerDumper -> ClassDefinition::getByName needs the Database (#12634)

mcop1 committed:
Apply php-cs-fixer changes

aashan10 committed:
Removed uninted deprecation warning

aashan10 committed:
Added dedicated method to get thumbnails in specific format (#12627)

* Added dedicated method to get thumbnails in specific format

* fix casing issue while generating thumbnail path

* Added public method to generate thumbnail images in given format

* Make custom format thumbnail generation deferred by default

* Removed unused code

* Cahnged the return type of the function

* Missing config case handled

* Update models/Asset/Thumbnail/ImageThumbnailTrait.php

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

* Update models/Asset/Thumbnail/ImageThumbnailTrait.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

* Fixed missing setConfig method issue

* Update models/Asset/Thumbnail/ImageThumbnailTrait.php

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>

* Update models/Asset/Thumbnail/ImageThumbnailTrait.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Reset thumbnail before setting a new format

* Added example to the documentatio

* Added assertions for getAsFormat method in thumbnails

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>
Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>
Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

mcop1 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Fix Phpstan errors

dvesh3 committed:
Deprecate Registering bundles through extensions.php (#12616)

* Deprecate Registering bundles through extensions.php - resolves #12274

* Deprecate Registering bundles through extensions.php - resolves #12274

* Deprecate Registering bundles through extensions.php - resolves #12274

* Deprecate Registering bundles through extensions.php - resolves #12274

* Deprecate Registering bundles through extensions.php - resolves #12274

* Deprecate Registering bundles through extensions.php - resolves #12274

* Fixed translation, added some more deprecations

* Deprecate Registering bundles through extensions.php - send deprecation info directly to output - resolves #12274

* Deprecate Registering bundles through extensions.php - some more deprecations - resolves #12274

Co-authored-by: Bernhard Rusch <bernhard.rusch@elements.at>

brusch committed:
Apply php-cs-fixer changes

aryaantony92 committed:
Disabled dragging and dropping assets from the Preview and List (#12681)

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

DuckThom committed:
Fix 'Only direct children' checkbox in variants tab

Fixes: #7726

Andreas-Koepke committed:
Remove locks for folders in Asset, Data Object and Document controllers. (#12680)

* Remove locks for folders in Asset, Data Object and Document controllers.

* Grab type from request - forgot line

storbahn committed:
Update README.md (#12690)

robertSt7 committed:
[Improvement]: Slow "object_url_slugs" queries / add indices (#12686)

* Improve: Add index for object_url_slugs queries

* Improve: Description for migration

* Update bundles/CoreBundle/Migrations/Version20220718162200.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

* Update bundles/CoreBundle/Migrations/Version20220718162200.php

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

aryaantony92 committed:
Prevents the calling of video processing method multiple times from the As… (#12692)

* Prevent calling of video processing method multiple times from the AssetUpdateTasksMessage f an exception is thrown

* Reverted the deletion of Log error message

dvesh3 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Core] Fix pimcore deprecations

jdreesen committed:
[Docs] Use map for static page config in nginx (#12676)

* [Docs] Use map for static page config in nginx

* Support static pages in the nginx demo config

* Align markdown table

* Add changelog entry

kingjia90 committed:
[Assets][Metadata]: Adds BC Layer for childs property in grid metadata tree (#12688)

* Fix: adds BC layer for metadata tree in grid config

* Docs: added TODO comment for the cleanup in 11

kingjia90 committed:
[Task]: Change broken link to documentation of headless chrome options (#12693)

* FIx: broken link to documentation of headless chrome

Alternatively https://github.com/spiritix/php-chrome-html2pdf#options

* Update bundles/AdminBundle/Resources/public/js/pimcore/settings/web2print.js

Co-authored-by: JiaJia Ji <kingjia90@gmail.com>

* FIx: broken link to documentation of headless chrome

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

robertSt7 committed:
Improve: Check if plugin.getClassName exist (#12696)

NiklasBr committed:
Add visual test for background in testWeb2print.html.twig (#12691)

also loads actual web2print settings for the testing (HeadlessChrome)

aashan10 committed:
Added a method to return publicly available full url path of the thumbnail (#12657)

* Added a method to return publicly available full url path of the thumbnail

* Fixed the changes according to review

* Update models/Asset/Thumbnail/ImageThumbnailTrait.php

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

* Added tests and public method for Assets

* Replaced deprecated assertion method

* Removed unused imports

Co-authored-by: Bernhard Rusch <brusch@users.noreply.github.com>

aryaantony92 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Fix Symfony deprecations - Command::execute() might add "int" as a native return type declaration in the future

blankse committed:
Avoid multiple __toString/getPath() calls in Video::getHtml5Code()

robertSt7 committed:
Improve: Wrap event listener mapping in a function (#12707)

dvesh3 committed:
Updated documentation for usage of url / path with link generators -  #12656

aryaantony92 committed:
Fix Symfony deprecations - Command::execute()

dvesh3 committed:
Fix Symfony deprecations - rest (#12705)

blankse committed:
Fix: Missing Video Thumbnails won't be generated (#12697)

* Fix: Missing Video Thumbnails won't be generated

* Use $this->getParameter()

* Remove getPath(), split in own PR

kingjia90 committed:
[Task]: Followup app logger flysystem (#12704)

* Fix: reintroduce the constant and adds a fallback

* Fix: added fallback to archive tasks

* Fix: added missing folder separator

* Fix: changed to fileExists

* Task: upgraded flysystem to v3

* Fix: wrong bundle

* Fix: reverting changes about bump flysystem to v3

details on https://github.com/pimcore/pimcore/pull/12704#issuecomment-1190198490

* Fix: added null check to file object fallback

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

Co-authored-by: mcop1 <89011527+mcop1@users.noreply.github.com>

mcop1 committed:
Apply php-cs-fixer changes

dvesh3 committed:
Fix Symfony deprecations - EventSubscriberInterface::getSubscribedEvents() might add "array" as a native return type (#12703)

* Fix Symfony deprecations - EventSubscriberInterface::getSubscribedEvents() might add "array" as a native return type declaration

* Changed Return Type for Non internal method

* Update lib/Sitemap/EventListener/SitemapGeneratorListener.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Update lib/Targeting/EventListener/DocumentTargetGroupListener.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

* Update lib/Targeting/EventListener/FullPageCacheCookieCleanupListener.php

Co-authored-by: Divesh Pahuja <divesh.pahuja@pimcore.com>

Co-authored-by: aryaantony92 <arya.antony@pimcore.com>
Co-authored-by: aryaantony92 <97134765+aryaantony92@users.noreply.github.com>

aryaantony92 committed:
Apply php-cs-fixer changes

dvesh3 committed:
[Docs] Fix upgrade notes link

fashxp committed:
Update POEditor translations

fashxp committed:
updated branch info

fashxp committed:
updated branch info

fashxp committed:
updated branch info

fashxp committed:
Merge remote-tracking branch 'origin/10.4' into 10.x

# Conflicts:
#	phpstan-baseline.neon

brusch committed:
[Sessions] Changed default value for `symfony.session.cookie_secure` to `auto` (#12711)

fashxp committed:
updated branch info

fashxp committed:
updated branch info

fashxp committed:
updated branch info

fashxp committed:
updated branch info

brusch committed:
updated dev demo instance

Commit:
Merge remote-tracking branch 'origin/patch-1' into patch-1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/pimcore/pimcore?pullRequest=11240) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/pimcore/pimcore?pullRequest=11240) before we can accept your contribution.<br/>**16** out of **17** committers have signed the CLA.<br/><br/>:white_check_mark: dvesh3<br/>:white_check_mark: dpfaffenbauer<br/>:white_check_mark: brusch<br/>:white_check_mark: mcop1<br/>:white_check_mark: DuckThom<br/>:white_check_mark: storbahn<br/>:white_check_mark: Andreas-Koepke<br/>:white_check_mark: aryaantony92<br/>:white_check_mark: aashan10<br/>:white_check_mark: kingjia90<br/>:white_check_mark: NiklasBr<br/>:white_check_mark: blankse<br/>:white_check_mark: fashxp<br/>:white_check_mark: robertSt7<br/>:white_check_mark: jdreesen<br/>:white_check_mark: passioneight<br/>:x: Michael Cuénez<br/><hr/>**Michael Cuénez** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pimcore/pimcore?pullRequest=11240) it.</sub>

passioneight commented:
<s>Can anyone list the steps for a proper rebase?

Here's what I did:
1. Cloned my repo.
2. Added pimcore/pimcore as remote
3. git fetch remote
4. Checkout patch-1
5. git rebase pimcore/11.x
6. Resolved merge conflict
7. Commit & Push
</s>

Turns out the issue was some setting in PhpStorm that merged the branch from origin after rebasing.
